tests/testthat/helper-functions.R

```diff
@@ -1,419 +1,442 @@
+    stop("Please load RSelenium: library(RSelenium)")
+  if (!"package:testthat" %in% search())
+    stop("Please load testthat: library(testthat)")
+  testDir <- find_test_path(dir)
+  # testthat::test_check assumes we are in path/to/animint/tests
+  old <- getwd()
+  on.exit(setwd(old), add = TRUE)
+  setwd(dirname(testDir))
+  # avoid weird errors if this function is called via testhat::check()
+  # https://github.com/hadley/testthat/issues/144
+  Sys.setenv("R_TESTS" = "")
+  testthat::test_check("animint2", filter = filter)
+}
+
+#' Kill child process(es) that may have been initiated in animint testing
+#'
+#' Read process IDs from a file and kill those process(es)
+#'
+#' @seealso \link{tests_run}
+#' @export
+tests_exit <- function() {
+  res <- stop_binary()
+  Sys.unsetenv("ANIMINT_BROWSER")
+  f <- file.path(find_test_path(), "pids.txt")
+  if (file.exists(f)) {
+    e <- try(readLines(con <- file(f), warn = FALSE), silent = TRUE)
+    if (!inherits(e, "try-error")) {
+      pids <- as.integer(e)
+      res <- c(res, tools::pskill(pids))
+    }
+    close(con)
+    unlink(f)
+  }
+  invisible(all(res))
+}
+
+#' Spawn a child R session that runs a 'blocking' command
+#'
+#' Run a blocking command in a child R session (for example a file server or shiny app)
+#'
+#' @param directory path that the  server should map to.
+#' @param port port number to _attempt_ to run server on.
+#' @param code R code to execute in a child session
+#' @return port number of the successful attempt
+run_servr <- function(directory = ".", port = 4848,
+                      code = "servr::httd(dir='%s', port=%d)") {
+  dir <- normalizePath(directory, winslash = "/", mustWork = TRUE)
+  cmd <- sprintf(
+    paste("write.table(Sys.getpid(), file='%s', append=T, row.name=F, col.names=F);", code),
+    file.path(find_test_path(), "pids.txt"), dir, port
+  )
+  system2("Rscript", c("-e", shQuote(cmd)), wait = FALSE)
+}
+
+# --------------------------
+# Functions that are used in multiple places
+# --------------------------
+
+stop_binary <- function() {
+  if (exists("pJS")) pJS$stop()
+  # these methods are really queries to the server
+  # thus, if it is already shut down, we get some arcane error message
+  e <- try({
+    remDr$closeWindow()
+    remDr$closeServer()
+  }, silent = TRUE)
+  TRUE
+}
+
+# find the path to animint's testthat directory
+find_test_path <- function(dir = ".") {
+  dir <- normalizePath(dir, winslash = "/", mustWork = TRUE)
+  if (!grepl("animint", dir, fixed = TRUE))
+    stop("animint must appear somewhere in 'dir'")
+  base_dir <- basename(dir)
+  if (!base_dir %in% c("animint2", "tests", "testthat"))
+    stop("Basename of dir must be one of: 'animint2', 'tests', 'testhat'")
+  ext_dir <- switch(base_dir,
+                    animint2 = "tests/testthat",
+                    tests = "testthat",
+                    testthat = "")
+  file.path(dir, ext_dir)
+}
+
+runtime_evaluate <- function(script=NULL){
+  
```

Comment:
please click Files Changed Tab and undo addition of all empty lines in functions

Reply:
space before and after <- are preferable style.
empty lines outside of the top level (in the middle of functions / blocks) should be avoided.